### PR TITLE
Softools connector

### PR DESCRIPTION
**Pull request by thduttonuk**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

### CODE REVIEW

**Comment by thduttonuk** [comment]
Docs say to add a label `certified-connector` but it cant seem to set it in Github

File: certified-connectors/Softools/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,879 @@
...
+            }
+          },
+          "401": {
+            "description": "",
```
Comment (sriyen-msft) [inline certified-connectors/Softools/apiDefinition.swagger.json:72]: @thduttonuk Please provide a valid description value.

File: certified-connectors/Softools/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,879 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "Response also includes all record data with field identifier as key",
+            "schema": {
+              "$ref": "#/definitions/Message"
+            }
+          },
+          "401": {
+            "description": "",
```
Comment (sriyen-msft) [inline certified-connectors/Softools/apiDefinition.swagger.json:123]: @thduttonuk Same here and everywhere in the swagger

File: certified-connectors/Softools/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,879 @@
...
+            "SourceType": 2,
+            "ActorType": 3,
+            "ActorId": "actorId",
+            "ParentId": null,
+            "Permissive": true
+          }
+        ],
+        "Id": "id",
+        "EditableAccessForUser": true,
+        "Tenant": "",
```
Comment (sriyen-msft) [inline certified-connectors/Softools/apiDefinition.swagger.json:644]: @thduttonuk Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/Softools/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,879 @@
...
+          "added": [
+            {
+              "Activity": "Item1",
+              "DateField2Example": {
+                "$date": 1466208000000
+              }
+            }
+          ],
+          "removed": [
+            "575fbf3cc99d15b138a08323"
```
Comment (sriyen-msft) [inline certified-connectors/Softools/apiDefinition.swagger.json:671]: @thduttonuk Can you please remove secret-look-a-like values from any property as we may receive security alerts from Microsoft Security and we need to provide explanation?

File: certified-connectors/Softools/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+            "required": "true",
+            "tabIndex": 2
+          },
+          "description": "The KEY for this API",
+          "displayName": "KEY",
+          "tooltip": "Provide your KEY"
+        }
+      }
+    },
+    "iconBrandColor": "#ee8426"
```
Comment (sriyen-msft) [inline certified-connectors/Softools/apiProperties.json:19]: @thduttonuk The `"publisher"` and `"stackOwner"` properties are missing.

Docs - [Add Metadata](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-3-add-metadata)

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/Softools/apiDefinition.swagger.json (hunk 4 above)
Comment (thduttonuk) [reply certified-connectors/Softools/apiDefinition.swagger.json:671]: I will fix all the comments today. This isn't a key and just a random id as an example, I will remove.

File: certified-connectors/Softools/apiDefinition.swagger.json (hunk 2 above)
Comment (thduttonuk) [reply certified-connectors/Softools/apiDefinition.swagger.json:123]: Done

File: certified-connectors/Softools/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,699 @@
...
+  "securityDefinitions": {
+    "APIKey": {
+      "type": "apiKey",
+      "in": "header",
+      "name": "apikey"
+    },
+    "Tenant": {
+      "type": "apiKey",
+      "in": "header",
+      "name": "tenant"
```
Comment (thduttonuk) [inline certified-connectors/Softools/apiDefinition.swagger.json:687]: I would like to set the `tenant` for all requests. This is the `AppId`. In the swagger docs it a Multiple Key example, is this supported in connectors? https://swagger.io/docs/specification/authentication/api-keys/

Comment (thduttonuk) [reply certified-connectors/Softools/apiDefinition.swagger.json:687]: Looks like this is not supported? https://powerusers.microsoft.com/t5/Connecting-To-Data/Multiple-API-Keys/td-p/402250 

For now the user will need to set the `tenant` header with each request.

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**